src/server.c

```diff
@@ -829,6 +831,35 @@ static bool taskReady(struct dqlite_node *d)
 	return d->running;
 }
 
+/* We take an exclusive lock on the file with this name in the raft directory
+ * while dqlite is running (between dqlite_node_start and dqlite_node_stop),
+ * and also during recovery (dqlite_node_recover_ext). This prevents corruption
+ * if two instances of dqlite try to use the same directory concurrently. */
+#define LOCK_FILENAME "dqlite-lock"
+
+static int acquire_dir(const char *dir, int *fd_out)
+{
+	char path[PATH_MAX];
+	snprintf(path, sizeof(path), "%s/%s", dir, LOCK_FILENAME);
+	int fd = open(path, O_RDONLY|O_CREAT, S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH);
```

Comment:
As discussed in our meeting I am not sure whether this approach vs open & unlink is better. Opening should work as well because only one process has to hold the lock, but I do not have a strong opinion.

Reply:
Discussed elsewhere---opening with `O_CREAT|O_EXCL` and then unlinking doesn't provide mutual exclusion (the second call to `open` creates a new inode, living alongside the old one which is no longer linked into the filesystem). And using `O_EXCL` without unlinking has the problem that if dqlite crashes the "lock" won't be released. So it seems like flock is the way to go here.